### DISCUSSION TITLE
issue: useFormContext. If Controller is unmounted and when mounted fieldArray does not contain defaultValue.

### DISCUSSION THREAD

**Discussion by alexanderleonenko** [react-hook-form/react-hook-form#6779] [Q&A]
### Version Number

7.17.2

### Codesandbox/Expo snack

https://codesandbox.io/s/fragrant-river-vg3cz

### Steps to reproduce

1. Go to [Codesandbox](https://codesandbox.io/s/fragrant-river-vg3cz).
2. Field are shown and filled with `defaultValues` (console's `form.watch()` log contains defined `testField` prop and `testProp` values).
2. Click on 'hide' button.
3. Click on 'show' button.
4. Fields aren't shown (console's `form.watch()` log contains `testField` field as  an empty array `[]`).

![fieldArray](https://user-images.githubusercontent.com/85992052/136943534-c72514e8-cfd4-4808-ba25-24af7f8d1792.gif)

### Expected behaviour

ArrayField `testField` prop contains `defaultValue`. It works in version `7.12.2`.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
why are you setting `shouldUnregister` to true?

> By default, an input value will be retained when input is removed. However, you can set shouldUnregister to true to unregister input during unmount.



![image](https://user-images.githubusercontent.com/10513364/136945037-871e7d32-f5e3-4af3-a6cf-76775d937e70.png)

**Comment by alexanderleonenko** [comment]
@bluebill1049 Because I don't want to have value for umnounted controls. But I want to get default values when I mounted it again. And it works with regular `fields`, but not with `arrayFields`.

**Comment by alexanderleonenko** [comment]
@bluebill1049 Actually it works with `arrayFields` also, but without `useFormContext()` (or in `7.12.2` version with `useFormContext()`).

**Comment by bluebill1049** [comment]
this is expected behavior, because `useFieldArray` has its own scope `useEffect`, by unmounting useFieldArray, it will unregister.

**Reply by bluebill1049** [reply]
code reference: https://github.com/react-hook-form/react-hook-form/blob/master/src/useFieldArray.ts#L293

**Reply by alexanderleonenko** [reply]
@bluebill1049 but when I mount `useFieldArray` again, should `useFieldArray` been registered again and take `defaultValues` again?

**Reply by alexanderleonenko** [reply]
@bluebill1049 
For example if there are `field` and `arrayField` and both are unregistered/registered, `field` takes `defaultValues`.
https://codesandbox.io/s/agitated-cdn-w1695

![2021-10-12 at 14 10 22](https://user-images.githubusercontent.com/85992052/136953579-03c46380-ee05-4e93-9b04-e70c96854c0c.gif)

**Comment by bluebill1049** [comment]
we are on 7.17.2 now, there are issues been resolving along the way.
